src/com/cloud/chatbot/token/TokenManager.java

```diff
@@ -0,0 +1,120 @@
+package com.cloud.chatbot.token;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.cloud.chatbot.exceptions.MissingInputException;
+
+
+
+/**
+ * A general parent Token manager providing children useful shared methods.
+ */
+public abstract class TokenManager {
+    protected List<Token> tokens = new ArrayList<>();
+
+    public TokenManager() {}
+
+    protected static List<Token> stringToTokens(String s) {
+        String trimmed = s.trim();
+
+        String word = "";
+        List<Token> tokens = new ArrayList<>();
+        for (int i = 0; i < trimmed.length(); i++) {
+            char c = trimmed.charAt(i);
+            if (c == ' ') {
+                if (word.length() > 0) {
+                    tokens.add(new Token(word));
+                    word = "";
+                }
+
+                tokens.get(tokens.size() - 1).incrementNextSpaces();
+            } else {
+                word += c;
+            }
+        }
+
+        if (word.length() > 0) {
+            tokens.add(new Token(word));
+        }
+
+        return tokens;
+    }
+
+    protected static String tokensToString(List<Token> tokens) {
+        String joined = "";
```

Comment:
Maybe can use a more meaningful name? It is a bit hard to infer what it is about simply by the variable name.

Reply:
The string starts empty but gets concatenated onto in subsequent lines, eventually being returned when done. In essence, it is a changing or accumulating temporary variable. If the logic gets more complex, I will consider changing the variable name or adding comments, maybe something like `joinedOutput` or documenting the protected method.